File: 1/CaesarCipherEncryptor.java
class CasesarCipherEncryptor {
	
	public static String caesarCipherEncryptor(String plaintext, int shift) {
		String ciphertext = "";
		
		for(char character : plaintext.toCharArray()) {
			ciphertext += ( (char)(97 + (character - 97 + shift) % 26) );
		}
		return ciphertext;
	}

	public static void main(String args[]) {
		System.out.println(caesarCipherEncryptor("abcdefghijklmnopqrstuvwxyz", 54));
	}
}

File: 1/js/src/CaesarCipherEncryptor.ts
function caesarCipher(plaintext: string, k: number): string {
	let ciphertext = "";

	k = k % 26;

	for(let i=0; i<plaintext.length; i++) {
		const newCharCode = 97 + ( plaintext.charCodeAt(i) - 97 + k) % 26;
		ciphertext += String.fromCharCode(newCharCode);
	}

	return ciphertext;
}

(function main() {
	const plaintext = "abcdefghijklmnopqrstuvwxyz";
	const k = 2;

	console.log(caesarCipher(plaintext, k))
})();
